adapters/httpapi_curl.c

```diff
@@ -1014,10 +1068,50 @@ HTTPAPI_RESULT HTTPAPI_CloneOption(const char* optionName, const void* value, co
             }
             else
             {
-                /*return OK when the certificate has been clones successfully*/
+                /*return OK when the certificate has been cloned successfully*/
                 result = HTTPAPI_OK;
             }
         }
+#ifdef USE_OPENSSL
+        else if (strcmp(OPTION_OPENSSL_PRIVATE_KEY_TYPE, optionName) == 0)
+        {
+            const OPTION_OPENSSL_KEY_TYPE type = **(const OPTION_OPENSSL_KEY_TYPE**)value;
```

Comment:
Crashes here as well: too many indirections.
It should be `const OPTION_OPENSSL_KEY_TYPE type = *(const OPTION_OPENSSL_KEY_TYPE*)value;`

Reply:
Fixed it.